src/codegen.cpp

```diff
@@ -2387,17 +2397,27 @@ static void mark_gc_use(const jl_cgval_t &v)
 static Value *make_jlcall(ArrayRef<const jl_cgval_t*> args, jl_codectx_t *ctx)
 {
     // the temporary variables are after all local variables in the GC frame.
-    CallInst *largs = CallInst::Create(prepare_call(jlcall_frame_func),
-            ConstantInt::get(T_int32, args.size()),
-            "",
-            /*InsertBefore*/ctx->ptlsStates);
+    // CallInst *largs = CallInst::Create(prepare_call(jlcall_frame_func),
+    //         ConstantInt::get(T_int32, args.size()),
+    //         "",
+    //         /*InsertBefore*/ctx->ptlsStates);
+    Value *largs = new AllocaInst(T_prjlvalue,
+#if JL_LLVM_VERSION >= 50000
+      0,
+#endif
+      ConstantInt::get(T_int32, args.size()), "jlcall", ctx->ptlsStates);
     int slot = 0;
     assert(args.size() > 0);
+    auto lifetime_start = Intrinsic::getDeclaration(jl_Module, Intrinsic::lifetime_start, {T_pprjlvalue});
+    builder.CreateCall(lifetime_start, {
```

Comment:
LLVM37+ syntax here, but elsewhere you seem to be maintaining LLVM33 support. Seems like the PR should just be consistent about dropping support?